src/bp/ui-studio/src/web/views/OneFlow/diagram/VariableTypesForm/PatternForm.tsx

```diff
@@ -0,0 +1,150 @@
+import { Tab, Tabs } from '@blueprintjs/core'
+import axios from 'axios'
+import sdk from 'botpress/sdk'
+import { Contents, lang, MoreOptions, MoreOptionsItems, RightSidebar } from 'botpress/shared'
+import _ from 'lodash'
+import React, { FC, Fragment, useEffect, useRef, useState } from 'react'
+
+import style from '../PromptForm/style.scss'
+
+import { getEntityId } from '.'
+
+interface Props {
+  customKey: string
+  contentLang: string
+  formData: sdk.NLU.EntityDefinition
+  close: () => void
+  deleteEntity: (entityId: string) => void
+  updateEntity: (originalId: string, entity: sdk.NLU.EntityDefinition) => void
+  updateFormItem: (entity) => void
+}
+
+const preparePattern = (pattern: string, matchCase?: boolean) => {
+  try {
+    let p = pattern || ''
+    if (!p.startsWith('^')) {
+      p = `^${p}`
+    }
+    if (!p.endsWith('$')) {
+      p = `${p}$`
+    }
+    return new RegExp(p, matchCase ? '' : 'i')
+  } catch (err) {
+    console.error('Pattern invalid', err)
+  }
+}
+
+const PatternForm: FC<Props> = ({
+  customKey,
+  contentLang,
+  formData,
+  close,
+  updateEntity,
+  deleteEntity,
+  updateFormItem
+}) => {
+  const originalEntity = useRef(formData)
+  const [showOptions, setShowOptions] = useState(false)
+  const [forceUpdate, setForceUpdate] = useState(false)
+  const [patternValid, setPatternValid] = useState(false)
+
+  const { pattern, matchCase, examples } = formData
+
+  useEffect(() => {
+    setForceUpdate(!forceUpdate)
+    originalEntity.current = formData
+  }, [customKey])
+
+  useEffect(() => {
+    if (_.isEqual(formData, originalEntity.current) || !formData.pattern || !patternValid) {
+      return
+    }
+
+    const newEntity = { ...formData, id: getEntityId(formData.name) }
+
+    updateEntity(originalEntity.current.id, newEntity)
+    originalEntity.current = newEntity
+  }, [formData])
+
+  useEffect(() => {
+    try {
+      new RegExp(pattern)
+      setPatternValid(true)
+    } catch (e) {
+      setPatternValid(false)
+    }
+  }, [pattern, matchCase, examples])
```

Comment:
why watch examples ? examples wont affect the pattern validity

Reply:
Right, and neither matchCase. Remnants of the old logic